### PR TITLE
Should close PreparedStatement and ResultSet in scan operations of Jd…

### PR DESCRIPTION
**Pull request by brfrn169**
…bcTransaction

I found we don't close PreparedStatement and ResultSet in scan operations of JdbcTransaction. This PR will fix this issue.

Please take a look!

### CODE REVIEW

File: src/main/java/com/scalar/db/storage/jdbc/JdbcService.java [hunk 1]
```diff
@@ -108,6 +109,40 @@ public Scanner scan(
         resultSet);
   }
 
+  public List<Result> scan(
+      Scan scan, Connection connection, Optional<String> namespace, Optional<String> tableName)
+      throws SQLException {
+    Utility.setTargetToIfNot(scan, namespacePrefix, namespace, tableName);
```
Comment (feeblefakie) [inline src/main/java/com/scalar/db/storage/jdbc/JdbcService.java:115]: I think `getScanner` and `scan` are pretty redundant, so should `scan` use `getScanner` (maybe it's not possible)  or should both delegate the common procedures to the same method?

**Review by feeblefakie — COMMENTED** [review]
Looking good! Left 1 clarifying question.

File: src/main/java/com/scalar/db/storage/jdbc/JdbcService.java (hunk 1 above)
Comment (brfrn169) [reply src/main/java/com/scalar/db/storage/jdbc/JdbcService.java:115]: Thank you for reviewing this! 

Sure, I will try that.

**Review by feeblefakie — APPROVED** [review]
LGTM!

**Pull request merged**